server/src/server/read/advisory.rs

```diff
@@ -0,0 +1,32 @@
+use actix_web::http::header::ContentType;
+use actix_web::http::StatusCode;
+use actix_web::{post, web, HttpResponse, Responder};
+use anyhow::anyhow;
+use csaf::Csaf;
+use std::fs;
+
+use crate::server::Error;
+use crate::AppState;
+
+use sha2::{Digest, Sha256};
+use trustify_ingestors as ingestors;
+
+#[utoipa::path(responses((status = 200, description = "Upload a file")))]
+#[post("/advisories")]
+pub async fn upload_advisory(
+    state: web::Data<AppState>,
+    payload: web::Payload,
+    content_type: Option<web::Header<ContentType>>,
+) -> Result<impl Responder, Error> {
+    let payload_bytes = payload.to_bytes().await?;
```

Comment:
I think we should put a not here that we need to fix this later. As this basically allocates memory twice the size of the document (the in memory blob and the parsed structured). I think it would make sense to (later) find a way how to parse the document from a stream.

Reply:
I added a TODO note.